src/Compilers/Core/Portable/Syntax/AbstractWarningStateMap.cs

```diff
@@ -87,5 +87,13 @@ public int CompareTo(WarningStateMapEntry other)
                 return this.Position - other.Position;
             }
         }
+
+        protected class WarningStateMapEntryComparer : System.Collections.IComparer
+        {
+            public int Compare(object x, object y)
+            {
+                return ((WarningStateMapEntry)x).CompareTo((WarningStateMapEntry)y);
```

Comment:
@davkean it's easily correctable and doesn't meet the stnadard of `IComparable`.  It doesn't leak today but doesn't mean it won't in the future.  The text of this comment is longer than the fix to make the code correct.